Bootloader/bootloader_core.c

```diff
@@ -0,0 +1,514 @@
+ * @brief read the boot mode in flash memory
+ * @param None
+ * @return boot_mode
+ ******************************************************************************/
+uint8_t LuosBootloader_GetMode(void)
+{
+    return LuosHAL_GetMode();
+}
+#endif
+
+#ifdef BOOTLOADER_CONFIG
+/******************************************************************************
+ * @brief DeInit peripherals in bootloader
+ * @param None
+ * @return None
+ ******************************************************************************/
+void LuosBootloader_DeInit(void)
+{
+    LuosHAL_DeInit();
+}
+#endif
+
+#ifdef BOOTLOADER_CONFIG
+/******************************************************************************
+ * @brief launch application from the bootloader
+ * @param None
+ * @return None
+ ******************************************************************************/
+void LuosBootloader_JumpToApp(void)
+{
+    LuosHAL_JumpToApp(APP_ADDRESS);
+}
+#endif
+
+/******************************************************************************
+ * @brief Save node id in flash
+ * @param None
+ * @return None
+ ******************************************************************************/
+void LuosBootloader_SaveNodeID(void)
+{
+    node_t *node     = Robus_GetNode();
+    uint16_t node_id = node->node_id;
+
+    LuosHAL_SaveNodeID(node_id);
+}
+
+#ifdef BOOTLOADER_CONFIG
+/******************************************************************************
+ * @brief Set node id with data saved in flash
+ * @param None
+ * @return None
+ ******************************************************************************/
+void LuosBootloader_SetNodeID(void)
+{
+    uint16_t node_id = LuosHAL_GetNodeID();
+    node_t *node     = Robus_GetNode();
+
+    node->node_id = node_id;
+}
+#endif
+
+#ifdef BOOTLOADER_CONFIG
+/******************************************************************************
+ * @brief Set node id with data saved in flash
+ * @param None
+ * @return None
+ ******************************************************************************/
+uint8_t LuosBootloader_IsEnoughSpace(uint32_t binary_size)
+{
+    uint32_t free_space = FLASH_END - APP_ADDRESS;
+    if (free_space > binary_size)
+    {
+        return 0x01;
+    }
+    else
+    {
+        return 0x00;
+    }
+}
+#endif
+
+#ifdef BOOTLOADER_CONFIG
+/******************************************************************************
+ * @brief process binary data received from the gate
+ * @param None 
+ * @return None
+ ******************************************************************************/
+void LuosBootloader_EraseMemory(void)
+{
+    LuosHAL_EraseMemory(APP_ADDRESS, nb_bytes);
+}
+#endif
+
+#ifdef BOOTLOADER_CONFIG
+/******************************************************************************
+ * @brief process binary data received from the gate
+ * @param None 
+ * @return None
+ ******************************************************************************/
+void LuosBootloader_ProcessData(void)
```

Comment:
This function is a kind of Luos_ReceiveData.
Why not use this one?
The case you are dealing with in bootloader could be a real usecase in other kind of services from users. Make it work with Luos_ReceiveData make Luos able to deal with it without having to make it manualy.

Reply:
Perhaps allowing to pull a bunch of data from a data_reception cession.

Reply:
Ok I get it. It's because Luos_ReceiveData can't manage more than 64Kb data.

Reply:
I think the use case is specific here because the function not only handles the copy in a buffer from luos messages but also the programming of this buffer in flash. I don't think a user would use Luos_ReceiveData in this case or if he does, he should write non trivial code to make it works in his special application case.

Reply:
In my point of view, it could be a common need. The idea is just to be able to get a chunk of a big receiving data and make. But it's OK to not make it like that on the BL because as you mention writing the actual chunk in flash et extremely slow and could lead to a Luos RAM overflow.
The ping pong way sounds good, keep it this way for now.